File: backend/app/schemas/product.py
"""商品数据模型文件：定义商品模块请求与响应结构。"""

from pydantic import BaseModel, ConfigDict, Field


class ProductBase(BaseModel):
    """商品基础字段模型：定义商品新增与编辑通用字段。"""

    name: str = Field(min_length=1, max_length=120)
    code: str = Field(min_length=1, max_length=60)
    category: str = 'software'
    spec_model: str = ''
    sale_price: float = 0
    unit: str = 'set'
    stock_qty: int = 0
    status: str = 'enabled'
    remark: str = ''


class ProductCreate(ProductBase):
    """商品新增请求模型。"""


class ProductUpdate(ProductBase):
    """商品编辑请求模型。"""

    id: int


class ProductOut(ProductBase):
    """商品响应模型。"""

    model_config = ConfigDict(from_attributes=True)

    id: int
    created_at: str
    updated_at: str

File: frontend/src/api/product.ts
// 商品接口文件：封装商品管理模块的列表、详情、新增、编辑、删除请求。
import http from './http'

// 商品列表项类型
export interface ProductItem {
  id: number
  name: string
  code: string
  category: string
  spec_model: string
  sale_price: number
  unit: string
  stock_qty: number
  status: string
  remark: string
  created_at: string
  updated_at: string
}

// 商品列表查询参数
export interface ProductListParams {
  keyword?: string
  category?: string
  status?: string
  page?: number
  page_size?: number
}

// 商品新增/编辑表单参数
export interface ProductFormPayload {
  id?: number
  name: string
  code: string
  category: string
  spec_model: string
  sale_price: number
  unit: string
  stock_qty: number
  status: string
  remark: string
}

// 获取商品列表（搜索 + 分页）
export const productListApi = (params: ProductListParams) => {
  return http.get('/api/product/list', { params })
}

// 获取商品详情
export const productDetailApi = (id: number) => {
  return http.get(`/api/product/detail/${id}`)
}

// 新增商品
export const productCreateApi = (payload: ProductFormPayload) => {
  return http.post('/api/product/create', payload)
}

// 编辑商品
export const productUpdateApi = (payload: ProductFormPayload) => {
  return http.put('/api/product/update', payload)
}

// 删除商品
export const productDeleteApi = (id: number) => {
  return http.delete(`/api/product/delete/${id}`)
}

